psycopg_c/psycopg_c/_psycopg/copy.pyx

```diff
@@ -52,8 +52,9 @@ def format_row_binary(
 
     if not tx._row_dumpers:
         tx._row_dumpers = PyList_New(rowlen)
-
     dumpers = tx._row_dumpers
+    if len(dumpers) != rowlen:
```

Comment:
Note to myself: The `len` call here costs ~10% additional runtime. If we can ensure, that `tx._row_dumpers` is always of a list type, we get the length check with `PyList_GET_SIZE` almost for free.